### PR TITLE
Fix: register profile comments parsing

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

While updating the register profile code I allowed to prepend comments to each register definition string.
E.g.: `"ctr DIER    .32 300 0  # (C64x+ only) Debug interrupt enable register\n"`
These comments can contain blank characters.

Later in the definition string parsing the definition is split at blank characters. So if the definition string contains a comment with blanks, the comment is split into X columns, although it should be a single column.

This let to a parsing failure and the register was not added to the register profile.

**Test plan**

Currently no arch is using these comments. PPC uses them in https://github.com/rizinorg/rizin/pull/2497 and TMS320 in https://github.com/rizinorg/rizin/pull/2805

**Closing issues**

...

### CODE REVIEW

File: librz/reg/profile.c [hunk 1]
```diff
@@ -272,7 +272,15 @@ static bool parse_reg_profile_str(RZ_OUT RzList *alias_list, RZ_OUT RzList *def_
 		if (RZ_STR_ISEMPTY(line)) {
 			continue;
 		}
-		toks = rz_str_split_duplist_n_regex(line, "[[:blank:]]+", 0, true);
+		if (index(line, '#')) {
+			RzList *line_and_cmt = rz_str_split_duplist_n_regex(line, "#", 0, true);
```
Comment (XVilka) [inline librz/reg/profile.c:276]: Can it be NULL? `strdup()` will segfault in that case.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
See also with ASAN:
```
  6/14 rizin:integration / cpu_platform_profiles          FAIL            1.25s   exit status 1
```

**Review by wargio — CHANGES_REQUESTED** [review]
add a test

**Comment by Rot127** [comment]
If you have no objections I'd like to merge https://github.com/rizinorg/rizin/pull/2805 first and add the test afterwards.

**Comment by Rot127** [comment]
https://github.com/rizinorg/rizin/pull/2809 should be merged before this. Or only https://github.com/rizinorg/rizin/pull/2805 and close this and #2809

**Review by XVilka — APPROVED** [review]

**Pull request merged**